src/Component.ts

```diff
@@ -151,37 +65,31 @@ class Component<T extends EventMap = DefaultEventMap> {
    * // https://github.com/naver/egjs-component/wiki/How-to-make-Component-event-design%3F
    * ```
    */
-  public trigger<K extends EventKey<T>>(eventName: K, ...params: any[]): boolean {
-    let handlerList = this._eventHandler[eventName] || [];
-    const hasHandlerList = handlerList.length > 0;
-
-    if (!hasHandlerList) {
-      return true;
-    }
-    const customEvent = params[0] || {};
-    const restParams = params.slice(1);
+  public trigger<K extends EventKey<T>>(event: K | _ComponentEvent<T[K], K>, ...params: EventTriggerParams<T, K> | void[]): this {
+    const eventName = event instanceof _ComponentEvent
+      ? event.eventType
+      : event;
 
-    // If detach method call in handler in first time then handler list calls.
-    handlerList = handlerList.concat();
+    const handlers = this._eventHandler[eventName] || [];
 
-    let isCanceled = false;
-
-    // This should be done like this to pass previous tests
-    (customEvent as any).eventType = eventName;
-    (customEvent as any).stop = () => { isCanceled = true; };
-    (customEvent as any).currentTarget = this;
-
-    let arg: any[] = [customEvent];
-
-    if (restParams.length >= 1) {
-      arg = arg.concat(restParams);
+    if (handlers.length <= 0) {
+      return this;
     }
 
-    handlerList.forEach(handler => {
-      handler.apply(this, arg);
-    });
+    if (event instanceof _ComponentEvent) {
+      event.currentTarget = this;
+
+      handlers.forEach((handler: (event: _ComponentEvent<T[K], K, this>) => any) => {
+        handler(event);
+      });
+    } else {
+      handlers.forEach(handler => {
+        // eslint-disable-next-line @typescript-eslint/no-unsafe-call
+        handler(...params);
+      });
+    }
 
-    return !isCanceled;
+    return this;
```

Comment:
How about keeping this part the same as before? In the case of ComponentEvent type, it returns whether the stop method is called as before, and if it is not, returns false, there will be no issues. Is there any reason to return this?

Reply:
Well I want to talk about that either :)
I just thought that as there's now isCanceled in ComponentEvent, it is no need to return boolean in trigger.

Reply:
An example of using the return value of trigger is as follows. The purpose of using stop in the `beforeX` event is to not execute the `X` event. In other words, it is necessary for the developer to design the event rather than the user that needs to cancel the event.
https://github.com/naver/egjs-component/wiki/How-to-make-Component-event-design%3F-(ko)#beforexxx-%EC%99%80-xxx-%EC%9D%B4%EB%B2%A4%ED%8A%B8

```js
if(!this.trigger("beforeX")){
    this.trigger("X");
}
```
In this case, it is difficult to handle with `isCancel`. How to solve it?

Reply:
You can more detail talk during the weekly meeting. :)

Reply:
okay, I'll talk about it during our weekly meeting :)

Reply:
```js
const event = new ComponentEvent("test", {});
this.trigger(event);
if (event.isCanceled()) {
}
```